ast-nodes/routine-declaration.dart

```diff
@@ -138,15 +138,20 @@ class RoutineDeclaration extends Declaration implements ScopeCreator {
     var routine = module.addRoutine(
       this.name,
       llvm.LLVMFunctionType(
-        this.returnType.getLlvmType(module),
+        this.returnType?.getLlvmType(module) ?? llvm.LLVMVoidTypeInContext(module.context),
         paramTypes,
         this.parameters.length,
         0,
       )
     );
 
-    // iterate over statements and add basic blocks
+    module.currentRoutine = routine;
 
+    for (var statement in this.body) {
+      statement.generateCode(module);
```

Comment:
Shouldn't all these function-calls return something? Why are we not doing anything with it?

Reply:
All these function calls are supposed to return an LLVM `Value`, but in some nodes it simply doesn't make sense. We just had to set a common interface on all AST nodes, it's okay if some of them will just return `null`

Reply:
So, basically, when we set `module.currentRoutine = routine`, then all the statements processed after it will be "added" to this routine?

Reply:
Well, if you add them, yes. We set this in order to have a reference to the `Value` object of the current routine which is necessary if we want to create a new basic block, and we want that for every statement